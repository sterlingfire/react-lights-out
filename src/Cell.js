import React from "react";
import "./Cell.css";

/** A single cell on the board.
 *
 * This has no state --- just three props:
 * 
 * - coord: a string 'y-x' where y is the row and x is the column location of
 *   the cell.
 *
 * - flipCellsAroundMe: a function rec'd from the board which flips this
 *      cell and the cells around of it
 *
 * - isLit: boolean, is this cell lit?
 *
 * This handles clicks --- by calling flipCellsAroundMe
 *
 **/

function Cell({ coord, flipCellsAroundMe, isLit }) {
  const classes = `Cell ${isLit ? "Cell-lit" : ""}`;
  return <td className={classes} onClick={() => flipCellsAroundMe(coord)} />;
}

export default Cell;
