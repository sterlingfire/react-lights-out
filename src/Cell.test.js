import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Cell from "./cell";

it("Should display a cell", function (){
  render(<Cell coord={"0-0"} cell-lit={true}/>);
});

it("Should display a cell", function (){
  expect(render(<Cell coord={"0-0"} cell-lit={true}/>).asFragment())
  .toMatchSnapshot();
});
