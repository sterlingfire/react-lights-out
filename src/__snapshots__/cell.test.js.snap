// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should display a cell 1`] = `
<DocumentFragment>
  <td
    class="Cell "
  />
</DocumentFragment>
`;
